cpp/src/table/legacy/device_table_row_operators.cuh

```diff
@@ -54,6 +56,41 @@ struct elements_are_equal {
     return static_cast<ColumnType const*>(lhs.data)[lhs_index] ==
            static_cast<ColumnType const*>(rhs.data)[rhs_index];
   }
+
+  template <typename ColumnType>
+  __device__ __forceinline__ 
+    typename std::enable_if_t<std::is_floating_point<ColumnType>::value, bool>
+    operator()(gdf_column const& lhs,
+                                             gdf_size_type lhs_index,
+                                             gdf_column const& rhs,
+                                             gdf_size_type rhs_index,
+                                             bool nulls_are_equal = false) {
+    if (nullable) {
+      bool const lhs_is_valid{gdf_is_valid(lhs.valid, lhs_index)};
```

Comment:
Does `nulls_are_equal` imply `NULL == NaN` is `true`? If so, I think I would restructure this a bit:

```
const ColumnType lhs_data = static_cast<ColumnType const*>(lhs.data)[lhs_index];
const ColumnType rhs_data = static_cast<ColumnType const*>(rhs.data)[rhs_index];
bool lhs_is_valid{not isnan(lhs_data)};
bool rhs_is_valid{not isnan(rhs_data)};

if (nullable) {
    lhs_is_valid ||= gdf_is_valid(lhs.valid, lhs_index);
    rhs_is_valid ||= gdf_is_valid(rhs.valid, rhs_index);
}

// If both values are invalid
if (not lhs_is_valid and not rhs_is_valid) {
    return nulls_are_equal;
}

// If only one value is null, they can never be equal
if (lhs_is_valid != rhs_is_valid) {
    return false;
}
   
return lhs_data == rhs_data;
```